internal/controller/stages/verification.go

```diff
@@ -377,6 +377,13 @@ func (r *reconciler) buildAnalysisRun(
 	}
 	lbls[kargoapi.StageLabelKey] = stage.Name
 	lbls[kargoapi.FreightLabelKey] = stage.Status.CurrentFreight.Name
+	// Add Promotion name if the AnalysisRun is triggered by Promotion.
+	// We can determine it by checking existence of Reverify key in annotations.
+	if _, ok := stage.GetAnnotations()[kargoapi.AnnotationKeyReverify]; !ok {
+		if stage.Status.LastPromotion != nil {
+			lbls[kargoapi.PromotionLabelKey] = stage.Status.LastPromotion.Name
+		}
+	}
```

Comment:
@krancour @hiddeco Not sure if it's okay to checking with the existence of Reverify key (Since verification is started after promotion is completed) 👀

Reply:
This should be ok.

One small warning... @rbreeze has detected that `LastPromotion` currently only gets set for Promotions that have _succeeded_ and doesn't get set otherwise.

I am working on fixing that. I don't think you should be too impacted by it anyway, since the verification process won't even kick off for a Promotion that failed. So this is just sort of an fyi.